pkg/controller/installation/core_controller.go

```diff
@@ -265,20 +262,8 @@ func add(c controller.Controller, r *ReconcileInstallation) error {
 				// Create occurs because we've created it, so we can safely ignore it.
 				return false
 			},
-			UpdateFunc: func(e event.UpdateEvent) bool {
-				if utils.IgnoreObject(e.ObjectOld) && !utils.IgnoreObject(e.ObjectNew) {
-					// Don't skip the removal of the "ignore" annotation. We want to
-					// reconcile when that happens.
-					return true
-				}
-				// Otherwise, ignore updates to objects when metadata.Generation does not change.
-				return e.ObjectOld.GetGeneration() != e.ObjectNew.GetGeneration()
```

Comment:
I have some concern that we are going to receive a lot more Reconcile triggers with these changes. Does this mean we'll now receive updates on status changes?

Reply:
I'm not sure, although we don't do this on any of our other watches as far as I'm aware - I think this predicate code was written very early on in the operator's life and might be based on some FUD. I'll investigate, though.

Reply:
FWIW, at least in my unit testing where I was monitoring updates, I didn't see any proliferation of excess Reconcile events - seemed like we were getting a single update for each resource

Reply:
I updated the common "AddNamespacedWatch" predicate logic to include this generation check so it will now be applied across all watches on namespaced resources. I think this is probably OK? It means we'll lose updates for when status / metadata fields change for namespaced resources.

I think the main time that we care about status updates are for watches on operator.tigera.io CRs which don't use this helper function.